File: backend/src/main/java/com/devmatch/backend/domain/user/service/AuthTokenService.java
package com.devmatch.backend.domain.user.service;

import com.devmatch.backend.domain.user.entity.User;
import com.devmatch.backend.standard.util.Ut;
import java.util.Map;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Service;

@Service
public class AuthTokenService {

  @Value("${custom.jwt.secretKey}")
  private String jwtSecretKey;

  @Value("${custom.accessToken.expirationSeconds}")
  private int accessTokenExpirationSeconds;

  String genAccessToken(User user) {
    long id = user.getId();
    String username = user.getUsername();//롬복
    String name = user.getNickName();//닉네임 가져오는 메서드

    return Ut.jwt.toString(
        jwtSecretKey,
        accessTokenExpirationSeconds,
        Map.of("id", id, "username", username, "name", name)
    );
  }

  Map<String, Object> payload(String accessToken) {
    Map<String, Object> parsedPayload = Ut.jwt.payload(jwtSecretKey, accessToken);

    if (parsedPayload == null) {
      return null;
    }

    int id = (int) parsedPayload.get("id");
    String username = (String) parsedPayload.get("username");
    String name = (String) parsedPayload.get("name");

    return Map.of("id", id, "username", username, "name", name);
  }
}

File: backend/src/main/java/com/devmatch/backend/domain/user/service/AuthTokenService.kt
package com.devmatch.backend.domain.user.service

import com.devmatch.backend.domain.user.entity.User
import com.devmatch.backend.standard.util.Ut
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Service

@Service
class AuthTokenService(
    @param:Value("\${custom.jwt.secretKey}")
    private val jwtSecretKey: String,

    @param:Value("\${custom.accessToken.expirationSeconds}")
    private val accessTokenExpirationSeconds: Int
) {
    fun genAccessToken(user: User): String {
        val id = user.id
        val username: String = user.username
        val nickname: String = user.nickname

        return Ut.jwt.toString(
            jwtSecretKey,
            accessTokenExpirationSeconds,
            mapOf<String, Any>("id" to id, "username" to username, "name" to nickname)
        )
    }

    fun payload(accessToken: String): Map<String, Any>? {
        val parsedPayload = Ut.jwt.payload(jwtSecretKey, accessToken)
            ?: return null

        val id = parsedPayload["id"] as Int
        val username = parsedPayload["username"] as String
        val nickname = parsedPayload["name"] as String

        return mapOf<String, Any>("id" to id, "username" to username, "name" to nickname)
    }
}